### ISSUE TITLE
System.Reflection.Emit NuGet has empty `lib` folders for nominally supported frameworks

### ISSUE THREAD

**Issue by Rabadash8820** [dotnet/runtime#83710]
### Description

The [System.Reflection.Emit](https://www.nuget.org/packages/System.Reflection.Emit) NuGet package nominally supports several frameworks, including various versions of .NET Framework, .NET Standard, .NET Core, Xamarin, etc. However, unzipping this package, it looks like all of those framework folders under `lib` are empty (except for a `_._` file) for all but a handful of frameworks. I'm not sure if this is a common situation across dotnet runtime libraries, but it seems weird to say a framework is supported but not actually contain DLLs built for it. Sounds like a bug to me, similar to #16322.

### Reproduction Steps

Download [System.Reflection.Emit](https://www.nuget.org/packages/System.Reflection.Emit) 4.7.0 from NuGet, unzip, and view the contents of the `lib` subfolders.

### Expected behavior

System.Reflection.Emit would have DLLs/PDBs for all supported frameworks.

### Actual behavior

System.Reflection.Emit only has DLLs/PDBs for the following supported frameworks:

- `netcore50`
- `netstandard1.1`
- `netstandard1.3`
- `netstandard2.0`

### Regression?

Don't think so... empty folders were present in versions 4.6.0 and 4.3.0 also (that's as far back as nuget.org displays).

### Known Workarounds

N/A

### Configuration

- .NET Standard 2.1 (Unity Mono)
- Windows 11 Pro
- x64
- Not sure if specific to this configuration
- Not on Blazor

### Other information

I discovered this issue while referencing `System.Reflection.Emit` in a Unity Engine project (using [UnityNuGet](https://github.com/xoofx/UnityNuGet)) that builds on a version of Mono compatible with .NET Standard 2.1. Because the folders are empty, the Unity Editor keeps showing me warnings like below. This is annoying and adds noise to "real" warnings, but also has me concerned that my netstandard2.1 project is apparently using a version of the DLL from previous frameworks.

```log
A meta data file (.meta) exists but its folder 'Packages/org.nuget.system.reflection.emit/netstandard2.1' can't be found, and has been created. Empty directories cannot be stored in version control, so it's assumed that the meta data file is for an empty directory in version control. When moving or deleting folders outside of Unity, please ensure that the corresponding .meta file is moved or deleted along with it.
```

**Comment by huoyaoyuan** [comment]
If a package provides only `_._` in the folder for a specific TFM, this typically means that it's included in the framework in box. You shouldn't ever need to install the package since the package doesn't provide anything additional.

> I'm not sure if this is a common situation across dotnet runtime libraries

This is a very common convention, especially for handling compatibility for packages that are in box on certain framework and not on others. It can be a bug for UnityNuGet not handling this.

**Comment by Rabadash8820** [comment]
Hi @huoyaoyuan, thanks for the information. Why exactly is the "dummy" `_._` file necessary then? Why are the empty TFM folders packed into the NuGet package if they don't actually add anything?

**Comment by huoyaoyuan** [comment]
> Why exactly is the "dummy" `_._` file necessary then?

For `System.Reflection.Emit`, it provides some files for `netstandard2.0`, but nothing for `netstandard2.1`. This means that it has something to add on 2.0, but those are included in box in 2.1 so nothing additional is required. Thus, the nuget package needs to explicitly provide nothing for `netstandard2.1` to prevent the fallback resolving to 2.0.

`.nupkg` is a zip file which doesn't have concept of folders. NuGet requires a folder present for supported TFM, so there must be some file under the path. The `_._` file is the conventional way of nupkg to create an empty folder.

**Comment by Rabadash8820** [comment]
Very interesting, thank you for clarifying! I will relay this information to the UnityNuGet authors then. I guess there's no actual issue here so feel free to close. 👌
